recipes/samurai/all/conanfile.py

```diff
@@ -0,0 +1,99 @@
+from conan import ConanFile
+from conan.errors import ConanInvalidConfiguration
+from conan.tools.build import check_min_cppstd
+from conan.tools.files import apply_conandata_patches, export_conandata_patches, get, copy
+from conan.tools.layout import basic_layout
+from conan.tools.scm import Version
+import os
+
+
+required_conan_version = ">=1.52.0"
+
+
+class PackageConan(ConanFile):
+    name = "samurai"
+    description = "Intervals coupled with algebra of set to handle adaptive mesh refinement and operators on it"
+    license = "BSD-3-Clause"
+    url = "https://github.com/conan-io/conan-center-index"
+    homepage = "https://github.com/hpc-maths/samurai"
+    topics = ("scientific computing", "adaptive mesh refinement", "header-only")
+    package_type = "header-library"
+    # Keep these or explain why it's not required for this particular case
+    settings = "os", "arch", "compiler", "build_type"
+    # Do not copy sources to build folder for header only projects, unless you need to apply patches
+    no_copy_source = True
+
+    @property
+    def _min_cppstd(self):
+        return "17"
+
+    @property
+    def _compilers_minimum_version(self):
+        return {
+            "Visual Studio": "16.9",
+            "msvc": "192",
+            "gcc": "12",
+            "clang": "13",
+            "apple-clang": "14",
+        }
+
+    def layout(self):
+        basic_layout(self, src_folder="src")
+
+    def requirements(self):
+        self.requires("cli11/2.3.2")
+        self.requires("fmt/9.1.0")
+        self.requires("highfive/2.7.1")
+        self.requires("pugixml/1.13")
+        self.requires("xsimd/10.0.0")
+        self.requires("xtensor/0.24.3")
+
+    def package_id(self):
+        self.info.clear()
+
+    def validate(self):
+        if self.settings.compiler.get_safe("cppstd"):
+            check_min_cppstd(self, self._min_cppstd)
+        minimum_version = self._compilers_minimum_version.get(str(self.settings.compiler), False)
+        if minimum_version and Version(self.settings.compiler.version) < minimum_version:
+            raise ConanInvalidConfiguration(
+                f"{self.ref} requires C++{self._min_cppstd}, which your compiler does not support."
+            )
+
+        # In case this library does not work in some another configuration, it should be validated here too
+        # if self.settings.os == "Windows":
+        #     raise ConanInvalidConfiguration(f"{self.ref} can not be used on Windows.")
```

Comment:
Feel free to remove unused code from the template that just adds noise :)

Reply:
removed